src/rebuild/srv.c

```diff
@@ -637,6 +701,39 @@ enum {
 	RB_BCAST_QUERY,
 };
 
+static void
+warn_for_slow_engine_updates(struct rebuild_global_pool_tracker *rgt)
+{
+	int    i;
+	double now    = ABT_get_wtime();
+	double tw     = now - rgt->rgt_last_warn; /* time since last warning logged */
+	bool   warned = false;
+
+	/* Throttle warnings to not more often than once per 2 minutes */
+	if (tw < 120)
+		return;
+
+	/* Warn for ranks not done and that haven't provided updates in a while (> 30 sec) */
+	for (i = 0; i < rgt->rgt_servers_number; i++) {
+		double   tu = now - rgt->rgt_servers_last_update[i];
```

Comment:
I am not sure but it looks 30 sec is too aggressive...

Reply:
target server reports progress every 2 second, so 30 sec here looks not too aggressive